modules/standard/IO.chpl

```diff
@@ -1302,10 +1302,9 @@ proc =(ref ret:file, x:file) {
 }
 
 /* Halt if a file is invalid */
-proc file.check() {
-  if(is_c_nil(_file_internal)) {
-    halt("Operation attempted on an invalid file");
-  }
+proc file.check() throws {
+  if is_c_nil(_file_internal) then
+    throw new FileNotFoundError("Operation attempted on an invalid file");
```

Comment:
The error ends up looking a bit odd this way, e.g.
```
$ ./util/start_test  test/library/standard/FileSystem/lydia/sameFile/givenNull.chpl
...
[Executing diff givenNull.good givenNull.exec.out.tmp]
1c1,3
< givenNull.chpl:7: error: halt reached - Operation attempted on an invalid file
---
> uncaught FileNotFoundError: No such file or directory (Operation attempted on an invalid file)
>   givenNull.chpl:7: thrown here
>   givenNull.chpl:7: uncaught here
```

Reply:
Ah that is a bit weird, I'll switch to a `SystemError`.